maxim/logger/components/generation.py

```
@@ -238,13 +265,20 @@ def add_message_(writer: LogWriter, id: str, message: GenerationRequestMessage):
                 "[MaximSDK] Invalid message. Must have 'content' and 'role' keys. We are skipping adding this message."
             )
             return
+        messages, attachments = parse_attachments_from_messages([message])
+        if len(attachments) > 0:
+            for attachment in attachments:
+                Generation.add_attachment_(writer, id, attachment)
         BaseContainer._commit_(
-            writer, Entity.GENERATION, id, "update", {"messages": [message]}
+            writer, Entity.GENERATION, id, "update", {"messages": messages}
```

Ruff (0.11.9)

Trailing comma missing

Add trailing comma

(COM812)